### ISSUE TITLE
Current bound_method() return of ContextualProperty.__get__ is broken

### ISSUE THREAD

**Issue by mshriver** [RonnyPfannschmidt/Sentaku#13]
Implementing an external_getter method for a sentaku.ContextualProperty and calling that method results in a RuntimeError.

Example: 

Class attribute is a ContextualProperty: https://github.com/ManageIQ/integration_tests/blob/master/cfme/base/__init__.py#L46

Method definition for this attribute:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/base/ui.py#L1224

Instantiating a Zone object and calling Zone.exists() results in RuntimeError:

```
----> 1 zone.exists()
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/sentaku/context.pyc in __get__(self, instance, owner)
    187         bound_method = implementation.__get__(instance, type(instance))
    188         with ctx.use(choice, frozen=True):
--> 189             return bound_method()
 
/home/mshriver/repos/integration_tests/cfme/base/ui.py in exists(self)
   1225 def exists(self):
   1226     try:
-> 1227         navigate_to(self, 'Details')
   1228         return True
   1229     except ZoneNotFound:
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in navigate(self, cls_or_obj, name, *args, **kwargs)
    102                 self.list_destinations(cls))
    103
--> 104         return nav(cls_or_obj, self).go(0, *args, **kwargs)
    105
    106     def list_destinations(self, cls_or_obj):
 
/home/mshriver/repos/integration_tests/utils/appliance/implementations/ui.pyc in go(self, _tries, *args, **kwargs)
    425         if not here:
    426             self.log_message("Prerequiesite Needed")
--> 427             self.prerequisite_view = self.prerequisite()
    428             self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
    429         if nav_args['use_resetter']:
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in __call__(self)
    191     def __call__(self):
    192         attr = self._get_attr(self.obj.obj)
--> 193         return self.obj.navigate_obj.navigate(attr, self.target)
    194
    195
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in navigate(self, cls_or_obj, name, *args, **kwargs)
    102                 self.list_destinations(cls))
    103
--> 104         return nav(cls_or_obj, self).go(0, *args, **kwargs)
    105
    106     def list_destinations(self, cls_or_obj):
 
/home/mshriver/repos/integration_tests/utils/appliance/implementations/ui.pyc in go(self, _tries, *args, **kwargs)
    425         if not here:
    426             self.log_message("Prerequiesite Needed")
--> 427             self.prerequisite_view = self.prerequisite()
    428             self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
    429         if nav_args['use_resetter']:
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in __call__(self)
    191     def __call__(self):
    192         attr = self._get_attr(self.obj.obj)
--> 193         return self.obj.navigate_obj.navigate(attr, self.target)
    194
    195
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in navigate(self, cls_or_obj, name, *args, **kwargs)
    102                 self.list_destinations(cls))
    103
--> 104         return nav(cls_or_obj, self).go(0, *args, **kwargs)
    105
    106     def list_destinations(self, cls_or_obj):
 
/home/mshriver/repos/integration_tests/utils/appliance/implementations/ui.pyc in go(self, _tries, *args, **kwargs)
    425         if not here:
    426             self.log_message("Prerequiesite Needed")
--> 427             self.prerequisite_view = self.prerequisite()
    428             self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
    429         if nav_args['use_resetter']:
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in __call__(self)
    163
    164     def __call__(self):
--> 165         return self.obj.navigate_obj.navigate(self.obj.obj, self.target)
    166
    167
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in navigate(self, cls_or_obj, name, *args, **kwargs)
    102                 self.list_destinations(cls))
    103
--> 104         return nav(cls_or_obj, self).go(0, *args, **kwargs)
    105
    106     def list_destinations(self, cls_or_obj):
 
/home/mshriver/repos/integration_tests/utils/appliance/implementations/ui.pyc in go(self, _tries, *args, **kwargs)
    425         if not here:
    426             self.log_message("Prerequiesite Needed")
--> 427             self.prerequisite_view = self.prerequisite()
    428             self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
    429         if nav_args['use_resetter']:
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in __call__(self)
    163
    164     def __call__(self):
--> 165         return self.obj.navigate_obj.navigate(self.obj.obj, self.target)
    166
    167
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/navmazing/__init__.pyc in navigate(self, cls_or_obj, name, *args, **kwargs)
    102                 self.list_destinations(cls))
    103
--> 104         return nav(cls_or_obj, self).go(0, *args, **kwargs)
    105
    106     def list_destinations(self, cls_or_obj):
 
/home/mshriver/repos/integration_tests/utils/appliance/implementations/ui.pyc in go(self, _tries, *args, **kwargs)
    425         if not here:
    426             self.log_message("Prerequiesite Needed")
--> 427             self.prerequisite_view = self.prerequisite()
    428             self.check_for_badness(self.step, _tries, nav_args, *args, **kwargs)
    429         if nav_args['use_resetter']:
 
/home/mshriver/repos/integration_tests/cfme/base/ui.py in prerequisite(self)
    242     def prerequisite(self):
    243         from utils.browser import ensure_browser_open
--> 244         ensure_browser_open(self.obj.appliance.server.address())
    245
    246     def step(self):
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/sentaku/context.pyc in __call__(self, *k, **kw)
    121         bound_method = implementation.__get__(
    122             self.instance, type(self.instance))
--> 123         with ctx.use(choice, frozen=ctx.strict_calls):
    124             return bound_method(*k, **kw)
    125
 
/usr/lib64/python2.7/contextlib.pyc in __enter__(self)
     15     def __enter__(self):
     16         try:
---> 17             return self.gen.next()
     18         except StopIteration:
     19             raise RuntimeError("generator didn't yield")
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/sentaku/context.pyc in use(self, *implementation_types, **kw)
     99
    100         with self.implementation_chooser.pushed(
--> 101                 implementation_types, frozen=_get_frozen(**kw)):
    102             yield self.impl
    103
 
/usr/lib64/python2.7/contextlib.pyc in __enter__(self)
     15     def __enter__(self):
     16         try:
---> 17             return self.gen.next()
     18         except StopIteration:
     19             raise RuntimeError("generator didn't yield")
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/sentaku/chooser.pyc in pushed(self, new, frozen)
     75     @contextmanager
     76     def pushed(self, new, frozen=False):
---> 77         self.current = Chooser.make(self.current, new, frozen)
     78         try:
     79             if len(chain(self.current)) > LIMIT:
 
/home/mshriver/repos/integration_tests/.cfme_tests/lib/python2.7/site-packages/sentaku/chooser.pyc in make(cls, current, elements, frozen)
     19         if current is not None and current.frozen:
     20             raise RuntimeError(
---> 21                 'further nesting of implementation choice has been disabled')
     22         if frozen:
     23             assert len(elements) == 1
 
RuntimeError: further nesting of implementation choice has been disabled

**Comment by mshriver** [comment]
If #12 is addressed with PR #14, this issue will become moot.

**Comment by RonnyPfannschmidt** [comment]
closing as missunderstaning
